cmd/create/cluster/cmd.go

```diff
@@ -2114,6 +2365,364 @@ func run(cmd *cobra.Command, _ []string) {
 			r.Reporter.Errorf("%s", err)
 			os.Exit(1)
 		}
+
+		if !isClusterAutoscalerSet && interactive.Enabled() {
+			isClusterAutoscalerSet, err = interactive.GetBool(interactive.Input{
+				Question: "Configure cluster-autoscaler",
+				Help:     "Set cluster-wide autoscaling configurations",
+				Default:  false,
+				Required: false,
+			})
+			if err != nil {
+				r.Reporter.Errorf("Expected a valid value for configure-cluster-autoscaler: %s", err)
+				os.Exit(1)
+			}
+		}
+
+		if isClusterAutoscalerSet {
+			if interactive.Enabled() && !isAutoscalerBalanceSimilarNodeGroupsSet {
+				autoscalerBalanceSimilarNodeGroups, err = interactive.GetBool(interactive.Input{
+					Question: "Balance similar node groups",
+					Help:     cmd.Flags().Lookup(autoscalerBalanceSimilarNodeGroupsFlag).Usage,
+					Default:  false,
+					Required: false,
```

Comment:
All of them are set of not required. If isClusterAutoscalerSet is true should they be required? Today the customer has the option to set `-y` and the non required fields would assume default or empty values. Has that been checked to cover if needed?

Reply:
From what I'm seeing, ``-y`` is only supposed to be used for automatically answering "yes" to confirm operations. Those questions are mainly asking for values rather than confirmation, so I think it shouldn't auto-set them to the defaults in this case

Reply:
`--mode auto -y` usually skips interactive mode as a whole :thinking: . Maybe there are some edge cases around it that I'm not familiar with that still goes into interactive, but I have the expectation that is not the case

cc @ciaranRoche @oriAdler Would you have some extra context here?

Reply:
hmm, also ``--mode auto`` mentions handling of AWS resources iiuc: https://github.com/openshift/rosa/blob/49ef038958eef794bea1d6e3f9065f9367b6e318/pkg/aws/flag.go#L41

would you say it's a blocker for the PR, or rather an enhancement that can be made in a follow-up?
Either way I'll look into that, for sure it can also help me with some of the manual testing that I'm doing :sweat_smile:

Reply:
As Guilherme said, the current behavior is that if the customer is running:
```
oadler@fedora:rosa (master)$ ./rosa create cluster --cluster-name oadler-test --mode auto -y
W: In a future release STS will be the default mode.
W: --sts flag won't be necessary if you wish to use STS.
W: --non-sts/--mint-mode flag will be necessary if you do not wish to use STS.
W: --mode is only valid for STS clusters
W: --mode is only valid for STS clusters
I: Creating cluster 'oadler-test'
I: To view a list of clusters and their status, run 'rosa list clusters'
I: Cluster 'oadler-test' has been created.
I: Once the cluster is installed you will need to add an Identity Provider before you can login into the cluster. See 'rosa create idp --help' for more information.

Name:                       oadler-test
ID:                         25lsfc4po2h6grn0kd3hgcguiur9ljik
External ID:                
Control Plane:              Customer Hosted
OpenShift Version:          
Channel Group:              stable
DNS:                        Not ready
AWS Account:                765374464689
API URL:                    
Console URL:                
Region:                     us-east-1
Multi-AZ:                   false
Nodes:
 - Control plane:           3
 - Infra:                   2
 - Compute:                 4
Network:
 - Type:                    OVNKubernetes
 - Service CIDR:            172.30.0.0/16
 - Machine CIDR:            10.0.0.0/16
 - Pod CIDR:                10.128.0.0/14
 - Host Prefix:             /23
Ec2 Metadata Http Tokens:   optional
State:                      pending (Preparing account)
Private:                    No
Created:                    Aug 17 2023 12:42:27 UTC
Details Page:               https://qaprodauth.console.redhat.com/openshift/details/s/2U729cFTwWK0TWexjDCLso2jvLa

I: To determine when your cluster is Ready, run 'rosa describe cluster -c oadler-test'.
I: To watch your cluster installation logs, run 'rosa logs install -c oadler-test --watch'.
```


It skips the interactive mode and sets the default values.
I would prefer to be consistent with the current ROSA CLI behavior unless there is a good reason to change it.

Reply:
ack. either way I can verify that running the following command:
```
./rosa create cluster --cluster-name odepaz-asg --autoscaler-scale-down-enabled --region us-west-1 --availability-zones us-west-1a --mode auto -y
```

does result with the defaults being automatically set for the cluster:
```
$ ocm get /api/clusters_mgmt/v1/clusters/<id>/autoscaler
{
  "kind": "ClusterAutoscaler",
  "href": "/api/clusters_mgmt/v1/clusters/<id>/autoscaler",
  "balance_similar_node_groups": false,
  "skip_nodes_with_local_storage": true,
  "log_verbosity": 1,
  "max_pod_grace_period": 0,
  "pod_priority_threshold": 0,
  "ignore_daemonsets_utilization": false,
  "resource_limits": {
    "max_nodes_total": 1000,
    "cores": {
      "min": 0,
      "max": 100
    },
    "memory": {
      "min": 0,
      "max": 4096
    }
  },
  "scale_down": {
    "enabled": true,
    "utilization_threshold": "0.500000"
  }
}
```